src/components/ImageView/ShowImageModal.vue

```diff
@@ -33,7 +33,7 @@ export default {
 			validator(imagesList) {
 				return (imagesList.length === 0)
 					? true
-					: imagesList.findIndex(({ basename, source }) => !(basename && source)) !== -1
+					: imagesList.findIndex(({ basename, source }) => !(basename && source)) === -1
```

Comment:
@luka-nextcloud could you take a look here? I'm not sure whether I got the logic of the code right, but it seemed to me that you wanted to check if all images in `imageList` have both `basename` and `source` set, right?

Reply:
The condition (with !==) returns false if we find any entry which doesn't have basename and/or source set. Please look carefully, the condition is evaluated on the value returned by findIndex. So it should be !== I think.

Reply:
Mh, not sure whether I can follow. You want this statement to return `true` if all children of `imagesList` have `basename` and `source` set, right? And if one of the children misses one of the two keys, you want it to return `false`, right?

So this was my test:

```js
[{ basename: 'a', source: 'b' }].findIndex(({ basename, source }) => !(basename && source)) === -1 // returns true
[{ basename: 'a' }].findIndex(({ basename, source }) => !(basename && source)) === -1 // returns false
```

vs.

```js
[{ basename: 'a', source: 'b' }].findIndex(({ basename, source }) => !(basename && source)) !== -1 // returns false
[{ basename: 'a' }].findIndex(({ basename, source }) => !(basename && source)) !== -1 // returns true
```

Reply:
To put your statement in words: `image.findIndex(...)` returns `-1` if no child is found where `basename` or `source` are unset. So the check for `image.findIndex(...) !== -1` is true if all children are valid, no?

Reply:
Let me propose yet another approach. You could either use the `.some` or `.every` functions on Array:
```suggestion
					: imagesList.every(image => image.basename && image.source)
```

I think you can even leave out the special handling for the empty list because `[].every(...)` is always true.

Reply:
Thanks, much easier to parse @max-nextcloud. I changed the code to your suggestion.